rllib/algorithms/dreamerv3/tests/test_dreamerv3.py

```diff
@@ -0,0 +1,200 @@
+        #  that - for now - merges the user provided model_dict (which only
+        #  contains settings that only affect the model, e.g. model_size)
+        #  with the AlgorithmConfig-wide settings that are relevant for the model
+        #  (e.g. `batch_size_B`).
+        # config.get_model_config()
+
+        num_iterations = 2
+
+        for _ in framework_iterator(config, frameworks="tf2"):
+            for env in ["FrozenLake-v1", "CartPole-v1", "ALE/MsPacman-v5"]:
+                print("Env={}".format(env))
+                config.environment(env)
+                algo = config.build()
+
+                for i in range(num_iterations):
+                    results = algo.train()
+                    print(results)
+
+                algo.stop()
+
+    def test_dreamerv3_dreamer_model_sizes(self):
+        """Tests, whether the different model sizes match the ones reported in [1]."""
+
+        # For Atari, these are the exact numbers from the repo ([3]).
+        # However, for CartPole + size "S" and "M", the author's original code will not
+        # match on the world model count. This is due to the fact that the author uses
+        # encoder/decoder nets with 5x1024 nodes (which corresponds to XL) regardless of
+        # the `model_size` settings (iff >="S").
+        expected_num_params_world_model = {
+            "XS_cartpole": 2435076,
+            "S_cartpole": 7493380,
+            "M_cartpole": 16206084,
+            "XS_atari": 7538979,
+            "S_atari": 15687811,
+            "M_atari": 32461635,
+        }
+
+        # All values confirmed against [3] (100% match).
+        expected_num_params_actor = {
+            # hidden=[1280, 256]
+            # hidden_norm=[256], [256]
+            # pi (2 actions)=[256, 2], [2]
+            "XS_cartpole": 328706,
+            "S_cartpole": 1051650,
+            "M_cartpole": 2135042,
+            "XS_atari": 329734,
+            "S_atari": 1053702,
+            "M_atari": 2137606,
+        }
+
+        # All values confirmed against [3] (100% match).
+        expected_num_params_critic = {
+            # hidden=[1280, 256]
+            # hidden_norm=[256], [256]
+            # vf (buckets)=[256, 255], [255]
+            "XS_cartpole": 393727,
+            "S_cartpole": 1181439,
+            "M_cartpole": 2297215,
+            "XS_atari": 393727,
+            "S_atari": 1181439,
+            "M_atari": 2297215,
+        }
+
+        config = (
+            dreamerv3.DreamerV3Config()
+            .framework("tf2", eager_tracing=True)
+            .training(
+                model={
+                    "batch_length_T": 16,
+                    "horizon_H": 5,
+                    "gamma": 0.997,
+                    "symlog_obs": True,
+                }
+            )
+        )
+
+        for model_size in ["S", "XS", "M"]:
+            config.model_size = model_size
+            config.model.update({
```

Comment:
can we use the `training` API to set this?

Reply:
yup, much better. fixed.